Code/GraphMol/ScaffoldNetwork/ScaffoldNetwork.h

```diff
@@ -0,0 +1,177 @@
+  }
+  bool operator!=(const RDKit::ScaffoldNetwork::NetworkEdge &o) const {
+    return (beginIdx != o.beginIdx) || (endIdx != o.endIdx) || (type != o.type);
+  }
+#ifdef RDK_USE_BOOST_SERIALIZATION
+ private:
+  friend class boost::serialization::access;
+  template <class Archive>
+  void serialize(Archive &ar, const unsigned int version) {
+    RDUNUSED_PARAM(version);
+    ar &beginIdx;
+    ar &endIdx;
+    ar &type;
+  }
+#endif
+};
+
+struct RDKIT_SCAFFOLDNETWORK_EXPORT ScaffoldNetwork {
+  std::vector<std::string> nodes;  ///< SMILES for the scaffolds
+  std::vector<unsigned>
+      counts;  ///< number of times each scaffold was encountered
+  std::vector<NetworkEdge> edges;  ///< edges in the network
+  ScaffoldNetwork(){};
+#ifdef RDK_USE_BOOST_SERIALIZATION
+  ScaffoldNetwork(const std::string &pkl) {
+    std::stringstream iss(pkl);
+    boost::archive::text_iarchive ia(iss);
+    ia >> *this;
+  }
+
+ private:
+  friend class boost::serialization::access;
+  template <class Archive>
+  void serialize(Archive &ar, const unsigned int version) {
+    RDUNUSED_PARAM(version);
+    ar &nodes;
+    ar &counts;
+    ar &edges;
+  }
+#endif
+};
+
+//! update an existing ScaffoldNetwork using a set of molecules
+template <typename T>
+void updateScaffoldNetwork(const T &mols, ScaffoldNetwork &network,
+                           const ScaffoldNetworkParams &params);
+
+//! create a new ScaffoldNetwork for a set of molecules
+template <typename T>
+ScaffoldNetwork createScaffoldNetwork(const T &mols,
+                                      const ScaffoldNetworkParams &params) {
+  ScaffoldNetwork res;
+  updateScaffoldNetwork(mols, res, params);
+  return res;
+}
+//! allows nodes to output nicely as strings
+inline std::ostream &operator<<(std::ostream &ostr,
+                                const RDKit::ScaffoldNetwork::EdgeType &e) {
+  switch (e) {
+    case RDKit::ScaffoldNetwork::EdgeType::Fragment:
+      ostr << "Fragment";
+      break;
+    case RDKit::ScaffoldNetwork::EdgeType::Generic:
+      ostr << "Generic";
+      break;
+    case RDKit::ScaffoldNetwork::EdgeType::GenericBond:
+      ostr << "GenericBond";
+      break;
+    case RDKit::ScaffoldNetwork::EdgeType::RemoveAttachment:
+      ostr << "RemoveAttachment";
+      break;
+    case RDKit::ScaffoldNetwork::EdgeType::Initialize:
+      ostr << "Initialize";
+      break;
+    default:
+      ostr << "UNKNOWN";
+      break;
+  }
+  return ostr;
+}
+//! allows edges to output nicely as strings
+inline std::ostream &operator<<(std::ostream &ostr,
+                                const RDKit::ScaffoldNetwork::NetworkEdge &e) {
+  ostr << "NetworkEdge( " << e.beginIdx << "->" << e.endIdx
+       << ", type:" << e.type << " )";
+  return ostr;
+}
+
+//! returns parameters for constructing scaffold networks using BRICS
+//! fragmentation
+RDKIT_SCAFFOLDNETWORK_EXPORT ScaffoldNetworkParams getBRICSNetworkParams();
```

Comment:
Is BRICS the only method?  I'm assuming this is controlled by the bond breakers?

Reply:
(smarts)

Reply:
It's non-trivial to construct the BRICS version, so this provides a simple call for creating it.
The default version uses a simple bond breaker SMARTS